hypothesis-python/src/hypothesis/strategies/_internal/strings.py

```diff
@@ -80,6 +86,18 @@ def do_draw(self, data):
 
         return chr(self.intervals[i])
 
+    def intersection(self, other):
+        assert isinstance(other, OneCharStringStrategy)
```

Comment:
Rebase as per other comment, and then we can remove this since I made `OneCharStringStrategy` constructible from an `IntervalSet` directly.